arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+        },
+        "ClaimType": {
+          "type": "string"
+        },
+        "ClaimValue": {
+          "type": "string"
+        },
+        "Rights": {
+          "type": "array",
+          "items": {
+            "enum": ["Manage", "Send", "Listen", "ManageNotificationHub"],
+            "type": "string"
+          }
+        },
+        "CreatedTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "ModifiedTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "Revision": {
+          "format": "int64",
+          "type": "integer"
+        }
+      }
+    },
+    "FeedbackProperties": {
+      "description": "Contains user facing view of Iot Hub Feedback Queue properties",
+      "type": "object",
+      "properties": {
+        "lockDurationAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "ttlAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "maxDeliveryCount": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "IotHubDescription": {
+      "description": "Contains user facing view of the Description of the IotHub",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the Iot Hub",
+          "type": "string"
+        },
+        "type": {
+          "description": "The type of the resource. It should include the resource provider namespace *and* the type of the entity. Microsoft.Devices/IotHubs.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "The location of the resource. This would be one amongst the supported Azure Geo Regions registered by the RP: West US | East US | North Central US | South Central US | West Europe | North Europe | East Asia | Southeast Asia | East US 2 | etc.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "A list of key value pairs that describe the resource. These tags can be used in viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource, and each tag must have a key no greater than 512 characters (and value no greater than 256 characters).",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "subscriptionid": {
+          "description": "The subscriptionId.",
+          "type": "string"
+        },
+        "resourcegroup": {
+          "description": "The resource group name uniquely identifies the resource group within the user subscriptionId. The resource group name must be no longer than 80 characters long, and must be alphanumeric characters (Char.IsLetterOrDigit()) and �-�, �_�, �(�, �)� and�.�. Note that the name cannot end with �.�",
+          "type": "string"
```

Comment:
You can add a regex pattern as a condition. Check this: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-resources/resources/2016-02-01/swagger/resources.json#L50